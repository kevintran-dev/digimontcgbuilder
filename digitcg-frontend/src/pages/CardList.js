import React from 'react'
import DigiCard from './DigiCard'

const CardList = ({ removeFromDeck, deckDiv, setDeckDiv, cardData, deckInfo, setdeckInfo }) => {
    console.log(cardData)
    return (
        <div className='DigiCard-list'>
            {Object.keys(cardData).map(data => (
                <DigiCard
                    key={cardData[data].cardnumber}
                    cardnumber={cardData[data].cardnumber}
                    artist={cardData[data].artist}
                    attribute={cardData[data].attribute}
                    card_sets={cardData[data].card_sets}
                    cardrarity={cardData[data].cardrarity}
                    color={cardData[data].color}
                    digi_type={cardData[data].digi_type}
                    dp={cardData[data].dp}
                    evolution_cost={cardData[data].evolution_cost}
                    image_url={cardData[data].image_url}
                    level={cardData[data].level}
                    maineffect={cardData[data].maineffect}
                    name={cardData[data].name}
                    play_cost={cardData[data].play_cost}
                    set_name={cardData[data].set_name}
                    sourceeffect={cardData[data].sourceeffect}
                    stage={cardData[data].stage}
                    type={cardData[data].type}
                    deckInfo={deckInfo}
                    setdeckInfo={setdeckInfo}
                    deckDiv={deckDiv}
                    setDeckDiv={setDeckDiv}
                    removeFromDeck={removeFromDeck}
                />
            ))}
        </div>
    )
}

export default CardList