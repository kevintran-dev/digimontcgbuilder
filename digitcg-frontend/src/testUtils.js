import React from "react";
import UserContext from "./auth/UserContext";

const UserProvider =
    ({ children, currentUsername = "testuser", currentUser = [], update = false }) => (
        <UserContext.Provider value={{ currentUser, currentUsername, update }}>
            {children}
        </UserContext.Provider>
    );

export { UserProvider };

