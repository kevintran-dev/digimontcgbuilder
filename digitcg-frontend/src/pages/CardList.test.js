import React from "react";
import { render } from "@testing-library/react";
import { UserProvider } from "../testUtils";
import CardList from "./CardList";

it("matches snapshot", function () {

    let card = [{ artist: "ryuda_1", attribute: "Vaccine", card_sets: ["ST-7: Starter Deck Gallantmon", "BT01-03: Release Special Booster Ver.1.0", "BTC-01: Booster Ultimate Evolution", "ST-7: Starter Deck Gallantmon", "BT-01: Booster New Evolution"], cardnumber: "BT1-009", cardrarity: "Common", color: "Red", digi_type: "Mini Dragon", dp: 3000, evolution_cost: 0, image_url: "https://images.digimoncard.io/images/cards/BT1-009.jpg", level: 3, maineffect: null, name: "Monodramon", play_cost: 2, set_name: "ST-7: Starter Deck Gallantmon", soureeffect: null, stage: "Rookie", type: "Digimon" }]

    const { asFragment } = render(
        <UserProvider>
            <CardList cardData={card} />
        </UserProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
});

