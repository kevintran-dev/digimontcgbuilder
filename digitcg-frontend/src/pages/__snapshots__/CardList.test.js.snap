// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="DigiCard-list"
  >
    <div>
      <img
        alt="Monodramon"
        class="DigiCard-img"
        src="https://images.digimoncard.io/images/cards/BT1-009.jpg"
      />
    </div>
  </div>
</DocumentFragment>
`;
